### PR TITLE
Feature/change date format and fix counter type

### PR DESCRIPTION
**Pull request by vov4ik-v**

### CODE REVIEW

File: src/main/java/com/spm/vasylyshyn/dto/DeviceDto.java [hunk 1]
```diff
@@ -25,6 +25,7 @@ public class DeviceDto {
     private Long frequency;
     private CType counterType;
     private Integer price;
+    private LocalDateTime createdDate;
 
 
 }
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/dto/DeviceDto.java]:
```suggestion
    private LocalDateTime createdDate;
}
```

File: src/main/java/com/spm/vasylyshyn/dto/DisplayCountDto.java [hunk 2]
```diff
@@ -5,10 +5,14 @@
...
+
 @Data
 @AllArgsConstructor
 @Builder
 public class DisplayCountDto {
     private Long id;
     private Long displayCount;
+    private LocalDateTime createdDate;
+
 }
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/dto/DisplayCountDto.java]:
```suggestion
    private LocalDateTime createdDate;
}
```

File: src/main/java/com/spm/vasylyshyn/dto/UserDto.java [hunk 3]
```diff
@@ -8,6 +8,8 @@
 import lombok.Data;
 import lombok.NoArgsConstructor;
 
+import java.time.LocalDateTime;
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/dto/UserDto.java]:
```suggestion
import java.time.LocalDateTime;
```

File: src/main/java/com/spm/vasylyshyn/enums/CType.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 package com.spm.vasylyshyn.enums;
 
 public enum CType {
-    GAS,WATER,LIGHT
+    GAS,WATER,LIGHT,NOT_SELECTED
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/enums/CType.java]: That's a good way to handle invalid enum values. I use `None` as the first member wherever my enum might not be initialized to a correct value, as a default. Also, we might not need it, if we should not have an ability to create a CType without a valid value, so I recoomend to review all the places that enum might be created. If we deserialize it, and it's null - the whole model should not be valid, and might be better null.

File: src/main/java/com/spm/vasylyshyn/enums/CType.java [hunk 5]
```diff
@@ -1,5 +1,5 @@
 package com.spm.vasylyshyn.enums;
 
 public enum CType {
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/enums/CType.java]:
```suggestion
public enum CounterType {
```

File: src/main/java/com/spm/vasylyshyn/model/Device.java [hunk 6]
```diff
@@ -44,22 +43,15 @@ public class Device {
...
-                        DateTimeFormatter
-                                .ofLocalizedDateTime( FormatStyle.SHORT )
-                                .withLocale(
-                                        new Locale( "ro" , "RO" )   // Romanian in Romania.
-                                )
-                );
+//    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSXXX")
+//    @JsonDeserialize(using = LocalDateDeserializer.class)
+//    @JsonSerialize(using = LocalDateSerializer.class)
+    private LocalDateTime createdDate;
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/model/Device.java]:
```suggestion
    private LocalDateTime createdDate;
```
Don't submit commented out code chunks

File: src/main/java/com/spm/vasylyshyn/model/User.java [hunk 7]
```diff
@@ -41,18 +47,28 @@ public class User implements UserDetails {
     @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.PERSIST, mappedBy = "owner")
     private List<Device> deviceList;
 
-    @JsonFormat(pattern = "yyyy-mm-dd HH:mm:ss")
-    @Column(updatable = false)
-    private LocalDateTime createdDate;
+    private String imageUrl;
+
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/model/User.java:51]:
```suggestion
```
I guess we don't use double line breaks between code lines unless for these cases:

Two blank lines should always be used in the following circumstances:

Between sections of a source file
Between class and interface definitions
One blank line should always be used in the following circumstances:
Between methods
Between the local variables in a method and its first statement
Before a block[ (see section 5.1.1)](https://www.cs.toronto.edu/km/istar/OME3/codeconv/CodeConventions.doc4.html#680) or single-line[ (see section 5.1.2)](https://www.cs.toronto.edu/km/istar/OME3/codeconv/CodeConventions.doc4.html#341) comment
Between logical sections inside a method to improve readability

File: src/main/java/com/spm/vasylyshyn/model/User.java [hunk 8]
```diff
@@ -41,18 +47,28 @@ public class User implements UserDetails {
...
     @Transient
     private Collection<? extends GrantedAuthority> authorities;
 
+//    @NotNull
+//    @Enumerated(EnumType.STRING)
+//    private AuthProvider provider;
+//
+//    private String providerId;
+
+
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/model/User.java:61]:
```suggestion
```

File: src/main/java/com/spm/vasylyshyn/repository/DisplayCountRepository.java [hunk 9]
```diff
@@ -13,15 +13,15 @@
...
-    @Query("select new com.spm.vasylyshyn.dto.DisplayCountDto(d.id,d.displayCount) from DisplayCount as d where d.id = max(d.id)")
-    Optional<DisplayCountDto>findDto(Device device);
+    @Query("select new com.spm.vasylyshyn.dto.DisplayCountDto(d.id,d.displayCount,d.createdDate) from DisplayCount as d where d.id = max(d.id)")
+    Optional<DisplayCountDto> findDto(Device device);
 
     List<DisplayCount> findDisplayCountsByDeviceAndCreatedDateGreaterThanEqualAndCreatedDateLessThanEqual(Device device, LocalDateTime startRange, LocalDateTime endRange);
 
 
-    @Query("select new com.spm.vasylyshyn.dto.DisplayCountDto(d.id,d.displayCount) from DisplayCount as d where d.device = :device")
+    @Query("select new com.spm.vasylyshyn.dto.DisplayCountDto(d.id,d.displayCount,d.createdDate) from DisplayCount as d where d.device = :device")
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/repository/DisplayCountRepository.java]:
```suggestion
    @Query("select new com.spm.vasylyshyn.dto.DisplayCountDto(d.id,d.displayCount, d.createdDate) from DisplayCount as d where d.device = :device")
```

File: src/main/java/com/spm/vasylyshyn/repository/UserRepository.java [hunk 10]
```diff
@@ -19,14 +19,14 @@ public interface UserRepository extends JpaRepository<User, Long> {
     Optional<User> findUserByUsername(String username);
     Optional<User> findUserByEmail(String email);
 
-    @Query("select new com.spm.vasylyshyn.dto.UserDto(u.id,u.username,u.firstName, u.lastName,u.phoneNumber,u.email,u.address) from User as u")
+    @Query("select new com.spm.vasylyshyn.dto.UserDto(u.id,u.username,u.firstName, u.lastName,u.phoneNumber,u.email,u.address,u.createdDate) from User as u")
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/repository/UserRepository.java]: fix  space for `, u.creted date` for this file

File: src/main/java/com/spm/vasylyshyn/request/UpdateDeviceRequest.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
+package com.spm.vasylyshyn.request;
+
+import com.spm.vasylyshyn.enums.CType;
+import com.spm.vasylyshyn.model.DisplayCount;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/request/UpdateDeviceRequest.java]:
```suggestion
```

File: src/main/java/com/spm/vasylyshyn/service/DeviceService.java [hunk 12]
```diff
@@ -37,7 +38,6 @@ public Device registerDevice(DeviceDto deviceDto){
     }
 
 //    public DeviceDto getDeviceDtoByNumber(Long deviceNumber) {
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/service/DeviceService.java]: Remove the commented out code

File: src/main/java/com/spm/vasylyshyn/service/UserService.java [hunk 13]
```diff
@@ -57,6 +58,7 @@ public User createUser(SignupRequest userIn) {
         user.setFirstName(userIn.getFirstname());
         user.setLastName(userIn.getLastname());
         user.setAvatarId(1);
+//        user.setProvider(AuthProvider.local);
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/service/UserService.java:61]:
```suggestion
```

File: src/main/java/com/spm/vasylyshyn/service/UserService.java [hunk 14]
```diff
@@ -135,7 +137,7 @@ public ApiResponse addDeviceToUser(Long deviceNumber, CType deviceType, String a
 //                                       String password,
                                        Principal principal) {
         User user = getUserByPrincipal(principal);
-        Device device = deviceService.registerDevice(new DeviceDto(deviceNumber,"",null,address,0L,deviceType,0));
+        Device device = deviceService.registerDevice(new DeviceDto(deviceNumber,"",null,address,0L,deviceType,0,LocalDateTime.now()));
```
Comment (BogdanDidukh2003) [inline src/main/java/com/spm/vasylyshyn/service/UserService.java]: Fix spaces between commas in the project

Comment (BogdanDidukh2003) [reply src/main/java/com/spm/vasylyshyn/service/UserService.java]: I just reformat the whole files for that reason

Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/service/UserService.java]: Fixed it

File: src/main/java/com/spm/vasylyshyn/service/UserService.java (hunk 13 above)
Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/service/UserService.java:61]: This need for feature update

File: src/main/java/com/spm/vasylyshyn/service/DeviceService.java (hunk 12 above)
Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/service/DeviceService.java]: Removed it

File: src/main/java/com/spm/vasylyshyn/repository/UserRepository.java (hunk 10 above)
Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/repository/UserRepository.java]: Fixed it

File: src/main/java/com/spm/vasylyshyn/model/User.java (hunk 7 above)
Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/model/User.java:51]: Ok

File: src/main/java/com/spm/vasylyshyn/model/User.java (hunk 8 above)
Comment (vov4ik-v) [reply src/main/java/com/spm/vasylyshyn/model/User.java:61]: It`s need for feature update

**Pull request merged**